reference/shipping_provider.yml

```diff
@@ -1807,16 +1721,32 @@ components:
               maxLength: 2
               example: US
             address_type:
-              description: Optional, defaults to residential
+              description: Defaults to residential. Optional.
               type: string
               enum:
                 - RESIDENTIAL
                 - COMMERCIAL
+            form_fields:
+              type: object
+              example:
+                '1234':
+                  id: '1234'
+                  value: checkbox_selection_1
+              description: Describes custom form field responses. Empty when no custom fields exist. 
+              properties:
+                '':
+                  type: object
+                  description: The ID and value of a custom form field response.
+                  properties:
+                    id:
```

Comment:
@bc-andreadao can you provide more clarity on what this ID is? an HTML ID, an ID related to the form_field being formally defined elsewhere...

Reply:
@slsriehl You raised a super interesting question, and I had a discussion with the team about it and we're going to change `id` to `name`, seeing how that's the convention in custom form fields elsewhere.

I'll update Andrea as code release progresses, although we're in a release freeze at the moment.

Reply:
OK, after all that, looks like it's actually going to go towards with keeping ID and value alone.

ID itself is the same to `shippingAddress` -> `id` found in https://developer.bigcommerce.com/api-reference/3c681bab90a14-get-form-fields.

Reply:
@bc-lorentz Thanks for getting back to me. Could you give me a description for the `value`? (I don't see it on https://developer.bigcommerce.com/api-reference/3c681bab90a14-get-form-fields) 

Also, just wanted to clarify, it seems like BC can only send **one** shipping address form field ID (`form_fields` is an object here) even though shipping addresses can have multiple form fields, right?

Reply:
Actual value is more related to:

https://developer.bigcommerce.com/api-reference/88829242e6459-get-a-shipping-address `form_fields` -> `value`

`id` is supposed to represent the global shared identification for the form field itself. `name` would be string label/representation of the field. `value` is the individual value of that field per the address.

Currently, multiple form fields will be represented as individual fields in an object, something like this:
```
"form_fields": {
  "1": "value_1",
  "2": "value_2"
}
```

^ Two form field details are supplied, for ids 1 and 2, with values "value_1" and "value_2".

Reply:
@bc-lorentz Thanks for the info. 

The `form_fields` schema you originally sent me over a slack message was this: 
```
"form_fields": {
   "27": {
      "id": "27",
      "value": "3"
   }
}
```

Based on what you just told me, should it be changed to:
```
"form_fields": {
   "27": "3"
}
```

Reply:
Yes, but might be best to put in a better value than `3`, maybe something like `selected_value`?

Reply:
@bc-lorentz Got it, thanks!